ggshield/config.py

```diff
@@ -1,322 +1,475 @@
+                if global_data and user_config.update_config(global_data):
+                    break
+            for local_config_path in LOCAL_CONFIG_PATHS:
+                local_data = load_yaml(local_config_path)
+                if local_data and user_config.update_config(local_data):
+                    break
+        # Env variables have priority over the local/global configs
+        user_config.set_env_gitguardian_urls()
+        return user_config
+
+    def add_ignored_match(self, secret: Dict) -> None:
+        """
+        Add secret to matches_ignore.
+        if it matches an ignore not in dict form, it converts it.
+        """
+        found = False
+        for i, match in enumerate(self.matches_ignore):
+            if isinstance(match, dict):
+                if match["match"] == secret["match"]:
+                    found = True
+                    # take the opportunity to name the ignored match
+                    if not match["name"]:
+                        match["name"] = secret["name"]
+            elif isinstance(match, str):
+                if match == secret["match"]:
+                    found = True
+                    self.matches_ignore[i] = secret
+            else:
+                raise click.ClickException("Wrong format found in ignored matches")
+        if not found:
+            self.matches_ignore.append(secret)
+
+
+@dataclass
+class AccountConfig:
+    account_id: int
+    token: str
+    type: str
+    token_name: str
+    expire_at: Optional[datetime]
+
+    def __post_init__(self) -> None:
+        if self.expire_at is not None and not isinstance(self.expire_at, datetime):
+            # Allow passing expire_at as an isoformat string
+            self.expire_at = datetime.fromisoformat(self.expire_at.replace("Z", "+00:00"))  # type: ignore
+
+    def to_dict(self) -> Dict:
+        data: Dict = custom_asdict(self, root=True)  # type: ignore
+        expire_at = data["expire_at"]
+        data["expire_at"] = expire_at.isoformat() if expire_at is not None else None
+        return data
+
+
+@dataclass
+class InstanceConfig:
+    account: AccountConfig  # Only handle 1 account per instance for the time being
+    url: str
+    name: Optional[str] = None
+    default_token_lifetime: Optional[timedelta] = None
+
+    @classmethod
+    def load(cls, data: Dict) -> "InstanceConfig":
+        accounts = data["accounts"]
+        assert (
+            len(accounts) == 1
+        ), "Each GitGuardian instance should have exactly one account"
+        data["account"] = AccountConfig(**data.pop("accounts")[0])
+        return cls(**data)
+
+    def to_dict(self) -> Union[List, Dict]:
+        data: Dict = custom_asdict(self, root=True)  # type: ignore
+        data["accounts"] = [data.pop("account")]
+        return data
+
+    @property
+    def expired(self) -> bool:
+        return (
+            self.account.expire_at is not None
+            and self.account.expire_at <= datetime.now()
```

Comment:
```suggestion
            and self.account.expire_at >= datetime.now()
```
also this assumes the datetime stored is unaware. By default when fetching the dt from the API they will be signed. This should either
* expect aware datetimes
* expect unaware datetimes
* expect anything but then utc-sign incoming unaware dt and compare it to `datetime.now(timezone.utc)`

Reply:
I disagree when the sign change, but agree with the rest :slightly_smiling_face:

Reply:
Agreed, I was waiting to see if the timezone awareness was requested.